### DISCUSSION TITLE
Switch HDMI on with AirPlay connected pi4

### DISCUSSION THREAD

**Discussion by Tiefflieger** [mikebrady/shairport-sync#1929] [General]
Hello, I would like to know if it's already implemented to be able to switch the HDMI on and off along with the AirPlay connection.
Background is that my pi is connected to my receiver via HDMI and it's always on. When I turn on my tv via ARC channel the receiver already starts on the raspberry pi input channel. This sets my tv for some case to select the internal speaker and I have to manually switch them back.
To encounter this I think it would help if I could switch the HDMI port along with the established AirPlay connection.

**Comment by mikebrady** [comment]
Thanks for this interesting question. I'm not familiar with ARC or eARC.

So, if you start up the Pi and then start Shairport Sync (assuming `disable_standby` is `"off"`) and don't play anything, does this TV/ARC misbehaviour occur?

**Comment by Tiefflieger** [comment]
Yeah I think it is due to the terminal output of my Raspberry Pi  on the hdmi port. This output seems to be always on, so the AVR sees this signal first and even if the tv gives the AVR the signal to switch to the TV Port on the AVR the TV internal speaker is set. I have to manually switch back to HDMI output on my tv.

I was playing around with chatgpt and made some research. Those were some ideas that might work:

Sadly my raspberry pi 4 with debian bookworm responds with `-1` to the command `sudo vcgencmd display_power 0`.
First I have to find another solution to switch my HDMI port. After that the following idea might be possible to switch the HDMI port on and off with:
```
LAST_STATUS="connected"

while true; do
  # Prüfen, ob eine `ESTABLISHED` Verbindung auf Port 7000 vorhanden ist
  CONNECTION_COUNT=$(netstat -an | grep :7000 | grep ESTABLISHED | wc -l)
  
  if [ "$CONNECTION_COUNT" -gt 0 ] && [ "$LAST_STATUS" = "disconnected" ]; then
    echo "AirPlay verbunden - HDMI einschalten"
    vcgencmd display_power 1
    LAST_STATUS="connected"
  elif [ "$CONNECTION_COUNT" -eq 0 ] && [ "$LAST_STATUS" = "connected" ]; then
    echo "AirPlay getrennt - HDMI ausschalten"
    vcgencmd display_power 0
    LAST_STATUS="disconnected"
  fi

  sleep 5
done
```

According to chatgpt it should be placed as a script here:
`sudo chmod +x /usr/local/bin/hdmi-control-docker.sh`

The File 
` /etc/systemd/system/hdmi-control-docker.service`
Should be created:
```
[Unit]
Description=HDMI Control based on Shairport-Sync connection in Docker
After=docker.service

[Service]
ExecStart=/usr/local/bin/hdmi-control-docker.sh
Restart=always

[Install]
WantedBy=multi-user.target
```
And activated/deactivated with:
```
sudo systemctl daemon-reload
sudo systemctl enable hdmi-control-docker.service
sudo systemctl start hdmi-control-docker.service
```


Before that I tried` docker logs -f <container-name>`, but I only got a connection message, no disconnect message:
`22.182363857 "rtsp.c:2896" Connection 8: AP2 PTP connection from 192.168.180.21:57773 ("user") to self at 192.168.180.55:7000.
`
So at the Moment for my case I only need a option to switch my HDMI Port on/off.
